amp-beta-tester.php

```diff
@@ -0,0 +1,202 @@
+	} else {
+		return $updates;
+	}
+
+	$amp_manifest = $updates->{$manifest_type}[ AMP_PLUGIN_FILE ];
+
+	$github_releases = get_amp_github_releases();
+
+	if ( is_array( $github_releases ) ) {
+
+		foreach ( $github_releases as $release ) {
+			if ( true === $release->prerelease ) {
+				$release_version = $release->tag_name;
+
+				// If there is a new release, let's see if there is a zip available for download.
+				if ( version_compare( $release_version, $amp_version, '>' ) ) {
+					$has_zip = false;
+
+					foreach ( $release->assets as $asset ) {
+						if ( 'amp.zip' === $asset->name ) {
+							$amp_manifest->package = $asset->browser_download_url;
+							$has_zip               = true;
+							break;
+						}
+					}
+
+					$amp_manifest->new_version = $release_version;
+					$amp_manifest->url         = $release->html_url;
+
+					unset( $updates->{$manifest_type}[ AMP_PLUGIN_FILE ] );
+
+					if ( $has_zip ) {
+						$updates->response[ AMP_PLUGIN_FILE ] = $amp_manifest;
+						break;
+					}
+				}
+			}
+		}
+	}
+
+	return $updates;
+}
+
+/**
+ * Fetch AMP releases from GitHub.
+ *
+ * @return array|null
+ */
+function get_amp_github_releases() {
+	$raw_response = wp_remote_get( 'https://api.github.com/repos/ampproject/amp-wp/releases' );
+	if ( is_wp_error( $raw_response ) ) {
+		return null;
+	}
+	return json_decode( $raw_response['body'] );
+}
+
+/**
+ * Replace the 'View version details' link with the link to the release on GitHub.
+ *
+ * @param string $file Plugin file.
+ * @param array  $plugin_data Plugin data.
+ */
+function replace_view_version_details_link( $file, $plugin_data ) {
+	$plugin_version = $plugin_data['Version'];
+
+	if ( is_pre_release( $plugin_version ) ) {
+		?>
+		<script>
+			document.addEventListener('DOMContentLoaded', function() {
+				const links = document.querySelectorAll("[data-slug='amp'] a.thickbox.open-plugin-details-modal");
+
+				links.forEach( (link) => {
+					link.className = 'overridden'; // Override class so that onclick listeners are disabled.
+					link.target = '_blank';
+					<?php
+					if ( isset( $plugin_data['url'] ) ) {
+						echo "link.href = '" . esc_url( $plugin_data['url'] ) . "';";
+					}
+					?>
+				} );
+			}, false);
+		</script>
+		<?php
+	}
+}
+
+/**
+ * Determine if the supplied version code is a prerelease.
+ *
+ * @param string $plugin_version Plugin version code.
+ * @return bool
+ */
+function is_pre_release( $plugin_version ) {
+	return (bool) preg_match( '/^\d+\.\d+\.\d+-(beta|alpha|RC)\d?$/', $plugin_version );
+}
+
+/**
+ * Displays the version code in the admin bar to act as a reminder that an unstable version
+ * of AMP is being used.
+ */
+function show_unstable_reminder() {
+	global $wp_admin_bar;
+
+	if ( is_pre_release( AMP__VERSION ) ) {
+		$args = [
+			'id'     => 'amp-beta-tester-admin-bar',
+			'title'  => sprintf( 'AMP v%s', AMP__VERSION ),
+			'parent' => 'top-secondary',
+			'href'   => admin_url( 'admin.php?page=amp-options' ),
```

Comment:
It needn't have a link even, or if it does, it could link to the plugins screen with "amp" being the search term to filter the list down to just the AMP plugin and the AMP Beta Tester plugin (among others), for example:

```suggestion
			'href'   => admin_url( 'plugins.php?s=amp&plugin_status=active' ),
```